src/components/common/Accordion/CustomAccordionFilter.tsx

```diff
@@ -0,0 +1,218 @@
+    marginRight: '20px',
+};
+
+export const CustomAccordionFilter = () => {
+    const [currentPage, setCurrentPage] = useState(1);
+    const [rowsPerPage, setRowsPerPage] = useState(10);
+    const [totalCount, setTotalCount] = useState(0);
+    const [reservedUsers, setReservedUsers] = useState<UserAndSkills[]>([]);
+    const [loading, setLoading] = useState(true);
+    const [snackbarOpen, setSnackbarOpen] = useState(false);
+    const [snackbarMessage, setSnackbarMessage] = useState('');
+    const [snackbarType, setSnackbarType] = useState('');
+    const [expanded, setExpanded] = useState<string | false>(false);
+
+    const { id } = useParams();
+    useEffect(() => {
+        const fetchReservedUsers = async (userId: string | undefined) => {
+        if(!userId) return null
+            try {
+                const response = await axios.get(`${apiUrl}/hr/${userId}?page=${currentPage}&limit=${rowsPerPage}`, { withCredentials: true });
+                const { users, totalCount } = response.data;
+                console.log(totalCount);
+                setReservedUsers(users);
+                setTotalCount(totalCount);
+                setLoading(false);
+            } catch (error) {
+                console.error(error);
+            }
+        };
+        setLoading(true);
+        fetchReservedUsers(id);
+    }, [id, currentPage, rowsPerPage]);
+
+    const handleShowCV = () => {
+        // @TODO: Logika obsługująca akcję "Pokaż CV"
+    };
+    const handleAction = async (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, email?: string, status?: string) => {
+        event.stopPropagation();
+        try {
+            await axios.patch(
+                `${apiUrl}${ENDPOINTS.setStatus}/${id}`,
+                {
+                    email: email,
+                    status: status,
+                },
+                { withCredentials: true }
+            );
+            setReservedUsers((prevUsers) => prevUsers.filter((user) => user.email !== email));
+                setSnackbarType('success');
+                setSnackbarOpen(true);
+            if (status === 'Dostępny') {
+                setSnackbarMessage('Kursant przestał być zarezerwowany');
+            } else if (status === 'Zatrudniony') {
+                setSnackbarMessage('Kursant został oznaczony jako zatrudniony');
+            }
+            const newTotalCount = totalCount - 1;
+            setTotalCount(newTotalCount);
+            const currentRowCount = reservedUsers.length;
+            if (currentRowCount < rowsPerPage) {
+                setRowsPerPage(currentRowCount);
+            }
+            const newPageCount = Math.ceil(newTotalCount / rowsPerPage);
+            if (currentPage > newPageCount) {
+                setCurrentPage(newPageCount);
+            }
+            if ((currentPage - 1) * rowsPerPage >= newTotalCount) {
+                setCurrentPage(currentPage - 1);
+            }
+        } catch (error: any) {
```

Comment:
może lepiej 'unknowon' zamiast any?

Reply:
tu miałam problem bo unknown wyrzuca mi błędy :(

Reply:
a np Error? Jak się nie da nic innego niż any to niech będzie any :)